### PR TITLE
Improve state handling in cases of memory allocation failure

### PR DESCRIPTION
**Pull request by AtariDreams**
Many lines of code assume that malloc will not fail. In cases where it does, sometimes the program does not know, and as a result, memory can leak and more disastrous consequences can happen before the program ultimately finds something is wrong and then calls exit();

### CODE REVIEW

File: cgi-bin/var.c [hunk 1]
```diff
@@ -553,15 +552,22 @@ cgi_add_variable(const char *name,	/* I - Variable name */
     if (!temp_vars)
       return;
 
-    form_vars  = temp_vars;
+    var = temp_vars + form_count;
+
+    if ((var->values = calloc((size_t)element + 1, sizeof(char *))) == NULL) {
```
Comment (michaelrsweet) [inline cgi-bin/var.c:557]: Coding style - open brace on its own line please.

File: scheduler/job.c [hunk 2]
```diff
@@ -1338,8 +1338,8 @@ cupsdContinueJob(cupsd_job_t *job)	/* I - Job */
   FilterLevel -= job->cost;
   job->cost = 0;
 
-  for (slot = 0; slot < 2; slot ++)
-    cupsdClosePipe(filterfds[slot]);
+  cupsdClosePipe(filterfds[0]);
```
Comment (michaelrsweet) [inline scheduler/job.c:1341]: Unrelated change.

File: tools/ippfind.c [hunk 3]
```diff
@@ -2503,9 +2503,12 @@ new_expr(ippfind_op_t op,		/* I - Operation */
       if (!strcmp(args[num_args], ";"))
         break;
 
-     temp->num_args = num_args;
-     temp->args     = malloc((size_t)num_args * sizeof(char *));
-     memcpy(temp->args, args, (size_t)num_args * sizeof(char *));
+    temp->num_args = num_args;
+    temp->args = malloc((size_t)num_args * sizeof(char *));
```
Comment (michaelrsweet) [inline tools/ippfind.c:2507]: Align "="

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Needs some cleanup.

**Comment by AtariDreams** [comment]
@michaelrsweet Hello?

File: cgi-bin/var.c [hunk 4]
```diff
@@ -553,15 +552,23 @@ cgi_add_variable(const char *name,	/* I - Variable name */
...
+    if ((var->values = calloc((size_t)element + 1, sizeof(char *))) == NULL)
+    {
+      /* Rollback changes*/
+      if (form_alloc == 0)
+        free(temp_vars);
+      return;
+    }
+    form_vars = temp_vars;
     form_alloc += 16;
+  } else {
```
Comment (michaelrsweet) [inline cgi-bin/var.c:566]: Coding style - should be:

```c
}
else
{
```

File: cups/globals.c [hunk 5]
```diff
@@ -180,8 +180,8 @@ DllMain(HINSTANCE hinst,		/* I - DLL module handle */
 static _cups_globals_t *		/* O - Pointer to global data */
 cups_globals_alloc(void)
 {
-  _cups_globals_t *cg = malloc(sizeof(_cups_globals_t));
-					/* Pointer to global data */
+  _cups_globals_t *cg =
+      calloc(1, sizeof(_cups_globals_t)); /* Pointer to global data */
```
Comment (michaelrsweet) [inline cups/globals.c:184]: calloc should be on first line, comment on second line

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Some minor coding style fixed needed, then it can be merged.

(and I've turned off Codacy scanning - too many false positives and that ridiculous MISRA crap)

**Comment by AtariDreams** [comment]
@michaelrsweet Done!

**Comment by AtariDreams** [comment]
@michaelrsweet All done!

File: cgi-bin/var.c [hunk 6]
```diff
@@ -544,7 +544,6 @@ cgi_add_variable(const char *name,	/* I - Variable name */
   {
     _cgi_var_t	*temp_vars;		/* Temporary form pointer */
```
Comment (michaelrsweet) [inline cgi-bin/var.c:546]: Two lines should separate var declarations and code at this level...

File: cgi-bin/var.c [hunk 7]
```diff
@@ -553,14 +552,24 @@ cgi_add_variable(const char *name,	/* I - Variable name */
     if (!temp_vars)
       return;
 
-    form_vars  = temp_vars;
+    var = temp_vars + form_count;
+
+    if ((var->values = calloc((size_t)element + 1, sizeof(char *))) == NULL)
+    {
+      /* Rollback changes*/
```
Comment (michaelrsweet) [inline cgi-bin/var.c:559]: Coding style:

```c
/*
 * Comment
 */
```

File: cups/globals.c [hunk 8]
```diff
@@ -180,8 +180,9 @@ DllMain(HINSTANCE hinst,		/* I - DLL module handle */
 static _cups_globals_t *		/* O - Pointer to global data */
 cups_globals_alloc(void)
 {
-  _cups_globals_t *cg = malloc(sizeof(_cups_globals_t));
-					/* Pointer to global data */
+  _cups_globals_t *cg =
+      calloc(1, sizeof(_cups_globals_t));
+  /* Pointer to global data */
```
Comment (michaelrsweet) [inline cups/globals.c:185]: Fix indentation

File: tools/ippeveprinter.c [hunk 9]
```diff
@@ -8069,7 +8069,7 @@ show_media(ippeve_client_t  *client)	/* I - Client connection */
     }
     else if (ready_sheets == 1)
       html_printf(client, ", 1 sheet</td></tr>\n");
-    else if (ready_sheets > 0)
+    else if (ready_sheets > 1)
```
Comment (michaelrsweet) [inline tools/ippeveprinter.c:8072]: Please don't include unrelated changed.

File: scheduler/cups-driverd.cxx [hunk 10]
```diff
@@ -2625,11 +2630,10 @@ load_ppds_dat(char   *filename,		/* I - Filename buffer */
       {
 	if ((ppd = (ppd_info_t *)calloc(1, sizeof(ppd_info_t))) == NULL)
 	{
-	  if (verbose)
-	    fputs("ERROR: [cups-driverd] Unable to allocate memory for PPD!\n",
-		  stderr);
-	  exit(1);
-	}
+    fputs("ERROR: [cups-driverd] Unable to allocate memory for PPD!\n",
```
Comment (michaelrsweet) [inline scheduler/cups-driverd.cxx:2633]: Please don't include unrelated changes.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
See comments...

File: scheduler/cups-driverd.cxx (hunk 10 above)
Comment (AtariDreams) [reply scheduler/cups-driverd.cxx:2633]: Because this regards memory, I would not call this an unrelated change.

**Review by michaelrsweet — APPROVED** [review]
